src/core/server/base.js

```diff
@@ -54,17 +52,52 @@ function getNoScriptStyles({ appName }) {
   return undefined;
 }
 
-function showErrorPage(res, status) {
-  let adjustedStatus = status;
-  let error = getErrorMsg(adjustedStatus);
-  if (!error) {
-    adjustedStatus = 500;
-    error = getErrorMsg(adjustedStatus);
+const appName = config.get('appName');
+
+function showErrorPage(res, createStore, status, error = {}) {
+  const store = createStore();
+
+  // Get SRI for deployed services only.
+  const sriData = (isDeployed) ? JSON.parse(
+    fs.readFileSync(path.join(config.get('basePath'), 'dist/sri.json'))
+  ) : {};
+
+  // Check the lang supplied by res.locals.lang for validity
+  // or fall-back to the default.
+  const lang = isValidLang(res.locals.lang) ?
+    res.locals.lang : config.get('defaultLang');
+  const dir = getDirection(lang);
+  store.dispatch(setLang(lang));
+  if (res.locals.clientApp) {
+    store.dispatch(setClientApp(res.locals.clientApp));
+  } else {
+    log.warn(`No clientApp for this ${status} error page`);
   }
-  return res.status(adjustedStatus).end(error);
-}
 
-const appName = config.get('appName');
+  store.dispatch({
+    type: REDUX_CONNECT_LOAD_FAIL,
```

Comment:
I'm dispatching this action because we already respond to it but arguably we should have our own action that does the same thing but isn't named after a redux action–as this one is almost certainly not caused by redux-connect.